### PR TITLE
[pymongo] Support newer msg requests

### PR DESCRIPTION
**Pull request by brettlangdon**
Newer versions of Mongo support sending a generic `OP_MSG` request instead of `OP_QUERY`, we should support these versions.

### CODE REVIEW

File: ddtrace/contrib/pymongo/parse.py [hunk 1]
```diff
@@ -79,14 +79,12 @@ def parse_msg(msg_bytes):
...
+    # NOTE[matt] inserts, updates and queries can all use this opcode
     db = None
     coll = None
-
-    offset = header_struct.size
-    cmd = None
     if op == 'query':
-        # NOTE[matt] inserts, updates and queries can all use this opcode
-
+        offset = header_struct.size
```
Comment (jd) [inline ddtrace/contrib/pymongo/parse.py]: That looked fine as it was, why moving it in each case?

Comment (brettlangdon) [reply ddtrace/contrib/pymongo/parse.py]: good catch, I was testing a bunch of approaches so this just happened to end up this way, I'll move back.

**Comment by brettlangdon** [comment]
When I added the newer versions of `pymongo` and `mongoengine` our existing tests failed.

**Comment by jd** [comment]
Right that probably mean our tests are ok then :)

**Comment by brettlangdon** [comment]
Haha, that is like "I'm not mad, just disappointed"

**Pull request merged**